### ISSUE TITLE
Build error with 0.9.99.0915

### ISSUE THREAD

**Issue by atsepkov** [gogits/gogs#3732]
- Gogs version (or commit ref): 0.9.99.0915
- Git version: 2.5.4 (Apple Git-61)
- Operating system: OSX deploying to Dokku on vagrant
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

```
[INFO] Setting version for gopkg.in/redis.v2 to e6179049628164864e6e84e973cfb56335748dea.
+ make build TAGS=sqlite cert pam
fatal: Not a git repository (or any of the parent directories): .git
go install "-v" -ldflags '-X "github.com/gogits/gogs/modules/setting.BuildTime=2016-10-05 06:03:23 UTC" -X "github.com/gogits/gogs/modules/setting.BuildGitHash="' -tags 'sqlite cert pam'
github.com/gogits/gogs/vendor/github.com/Unknwon/cae
github.com/gogits/gogs/vendor/github.com/Unknwon/cae/zip
github.com/gogits/gogs/vendor/github.com/Unknwon/com
github.com/gogits/gogs/vendor/github.com/go-macaron/inject
github.com/gogits/gogs/vendor/gopkg.in/ini.v1
github.com/gogits/gogs/vendor/gopkg.in/macaron.v1
github.com/gogits/gogs/vendor/github.com/go-macaron/binding
github.com/gogits/gogs/vendor/github.com/go-macaron/cache
github.com/gogits/gogs/vendor/github.com/go-macaron/captcha
github.com/gogits/gogs/vendor/github.com/go-macaron/session
github.com/gogits/gogs/vendor/github.com/go-macaron/csrf
github.com/gogits/gogs/vendor/github.com/klauspost/cpuid
github.com/gogits/gogs/vendor/github.com/klauspost/compress/flate
github.com/gogits/gogs/vendor/github.com/klauspost/crc32
github.com/gogits/gogs/vendor/github.com/klauspost/compress/gzip
github.com/gogits/gogs/vendor/github.com/go-macaron/gzip
github.com/gogits/gogs/vendor/github.com/Unknwon/i18n
github.com/gogits/gogs/vendor/golang.org/x/text/internal/tag
github.com/gogits/gogs/vendor/golang.org/x/text/language
github.com/gogits/gogs/vendor/github.com/go-macaron/i18n
github.com/gogits/gogs/vendor/github.com/go-macaron/toolbox
github.com/gogits/gogs/vendor/github.com/go-xorm/core
github.com/gogits/gogs/vendor/github.com/go-xorm/xorm
github.com/gogits/gogs/vendor/github.com/mcuadros/go-version
github.com/gogits/gogs/vendor/github.com/gogits/git-module
github.com/gogits/gogs/vendor/github.com/gogits/go-gogs-client
github.com/gogits/gogs/vendor/github.com/go-sql-driver/mysql
github.com/gogits/gogs/vendor/github.com/gogits/chardet
github.com/gogits/gogs/modules/log
github.com/gogits/gogs/vendor/github.com/bradfitz/gomemcache/memcache
github.com/gogits/gogs/vendor/github.com/go-macaron/cache/memcache
github.com/gogits/gogs/vendor/gopkg.in/bufio.v1
github.com/gogits/gogs/vendor/gopkg.in/redis.v2
github.com/gogits/gogs/vendor/github.com/go-macaron/cache/redis
github.com/gogits/gogs/vendor/github.com/go-macaron/session/redis
github.com/gogits/gogs/modules/bindata
github.com/gogits/gogs/modules/user
github.com/gogits/gogs/vendor/github.com/strk/go-libravatar
github.com/gogits/gogs/modules/setting
github.com/gogits/gogs/modules/base
github.com/gogits/gogs/vendor/github.com/satori/go.uuid
github.com/gogits/gogs/models/migrations
github.com/gogits/gogs/vendor/gopkg.in/asn1-ber.v1
github.com/gogits/gogs/vendor/gopkg.in/ldap.v2
github.com/gogits/gogs/modules/auth/ldap
github.com/gogits/gogs/vendor/github.com/msteinert/pam
github.com/gogits/gogs/modules/auth/pam
github.com/gogits/gogs/vendor/github.com/issue9/identicon
github.com/gogits/gogs/modules/avatar
github.com/gogits/gogs/modules/httplib
github.com/gogits/gogs/vendor/golang.org/x/net/html/atom
github.com/gogits/gogs/vendor/golang.org/x/net/html
github.com/gogits/gogs/vendor/github.com/jaytaylor/html2text
github.com/gogits/gogs/vendor/gopkg.in/gomail.v2
github.com/gogits/gogs/modules/mailer
github.com/gogits/gogs/vendor/github.com/microcosm-cc/bluemonday
github.com/gogits/gogs/vendor/github.com/shurcooL/sanitized_anchor_name
github.com/gogits/gogs/vendor/github.com/russross/blackfriday
github.com/gogits/gogs/modules/markdown
github.com/gogits/gogs/modules/process
github.com/gogits/gogs/modules/sync
github.com/gogits/gogs/modules/template/highlight
github.com/gogits/gogs/vendor/github.com/lib/pq/oid
github.com/gogits/gogs/vendor/github.com/lib/pq
github.com/gogits/gogs/vendor/github.com/mattn/go-sqlite3
github.com/gogits/gogs/vendor/github.com/nfnt/resize
github.com/gogits/gogs/vendor/github.com/sergi/go-diff/diffmatchpatch
github.com/gogits/gogs/vendor/golang.org/x/crypto/curve25519
github.com/gogits/gogs/vendor/golang.org/x/crypto/ed25519/internal/edwards25519
github.com/gogits/gogs/vendor/golang.org/x/crypto/ed25519
github.com/gogits/gogs/vendor/golang.org/x/crypto/ssh
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/internal/identifier
github.com/gogits/gogs/vendor/golang.org/x/text/transform
github.com/gogits/gogs/vendor/golang.org/x/text/encoding
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/internal
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/charmap
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/japanese
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/korean
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/simplifiedchinese
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/traditionalchinese
github.com/gogits/gogs/vendor/golang.org/x/text/internal/utf8internal
github.com/gogits/gogs/vendor/golang.org/x/text/runes
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/unicode
github.com/gogits/gogs/vendor/golang.org/x/text/encoding/htmlindex
github.com/gogits/gogs/vendor/golang.org/x/net/html/charset
github.com/gogits/gogs/models
# github.com/gogits/gogs/models
models/access.go:84: cannot use x (type *xorm.Engine) as type Engine in argument to accessLevel:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/access.go:94: cannot use x (type *xorm.Engine) as type Engine in argument to hasAccess:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/access.go:245: cannot use x (type *xorm.Engine) as type Engine in argument to r.recalculateAccesses:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/action.go:194: cannot use x (type *xorm.Engine) as type Engine in argument to newRepoAction:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/action.go:217: cannot use x (type *xorm.Engine) as type Engine in argument to renameRepoAction:


r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session


models/action.go:568: cannot use x (type *xorm.Engine) as type Engine in argument to transferRepoAction:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session


rewant Where(interface {}, ...interface {}) *xorm.Session
models/action.go:586: cannot use x (type *xorm.Engine) as type Engine in argument to mergePullRequestAction:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/issue.go:154: cannot use x (type *xorm.Engine) as type Engine in argument to issue.loadAttributes:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/issue.go:231: cannot use x (type *xorm.Engine) as type Engine in argument to i.hasLabel:
r*xorm.Engine does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/issue.go:312: cannot use e (type *xorm.Session) as type Engine in argument to issue.getLabels:
r*xorm.Session does not implement Engine (wrong type for Where method)
rehave Where(string, ...interface {}) *xorm.Session
rewant Where(interface {}, ...interface {}) *xorm.Session
models/issue.go:312: too many errors
github.com/gogits/gogs/vendor/github.com/Unknwon/paginater
github.com/gogits/gogs/vendor/gopkg.in/editorconfig/editorconfig-core-go.v1
github.com/gogits/gogs/vendor/github.com/gogits/cron
github.com/gogits/gogs/vendor/github.com/urfave/cli
Makefile:31: recipe for target 'build' failed
make: *** [build] Error 2
remote: The command '/bin/sh -c ./docker/build.sh' returned a non-zero code: 2
To dokku@my.dokku:edit
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@my.dokku:edit'
```

This error looks very similar to https://github.com/gogits/gogs/issues/3712 to me, but the latest commit should have fixed it, trying to understand why it's still happening for me.
# Description

Trying to get gogs up and running, for the first time, here are the steps I took:
- git clone https://github.com/gogits/gogs.git (git log latest commit: 2bec8a4f1ef6b48220a0010c1482e561abd55810)
- git remote add dokku dokku@dokku.me:gogs
- git push dokku master (failed due to macaron version)
- manually applied the fix (https://github.com/gogits/gogs/pull/3727/files)
- git add/commit/push again
- encounter the error shown above

**Comment by LefsFlarey** [comment]
Do try running `go get -u all`. You are not running the latest version of `go-xorm`.

**Comment by atsepkov** [comment]
This isn't my local go install. If I'm understanding the setup correctly, go gets auto-downloaded by Docker, from the logs it looks like it ends up being downloaded into /tmp:
```
+ cd /tmp/go/src/github.com/Masterminds/glide
+ make build
go build -o glide -ldflags "-X main.version=0.10.2" glide.go
+ go install
+ mkdir -p /tmp/go/src/github.com/gogits/
+ ln -s /app/gogs/ /tmp/go/src/github.com/gogits/gogs
+ cd /tmp/go/src/github.com/gogits/gogs
+ glide install
```
Quick search for above text pointed me to gogs/docker/build.sh in your repo, so I added your recommended command before make, and now receiving the following error instead:
```
Executing busybox-1.24.2-r0.trigger
OK: 371 MiB in 62 packages
+ git clone -b 0.10.2 https://github.com/Masterminds/glide /tmp/go/src/github.com/Masterminds/glide
Cloning into '/tmp/go/src/github.com/Masterminds/glide'...
Note: checking out '2d0eb388e16d833ba3c55068335b3577b6b09ff0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

+ cd /tmp/go/src/github.com/Masterminds/glide
+ go get -u all
package cmd/internal/rsc.io/arm/armasm: directory "/usr/lib/go/src/cmd/internal/rsc.io/arm/armasm" is not using a known version control system
package cmd/internal/rsc.io/x86/x86asm: directory "/usr/lib/go/src/cmd/internal/rsc.io/x86/x86asm" is not using a known version control system
remote: The command '/bin/sh -c ./docker/build.sh' returned a non-zero code: 1
To dokku@my.dokku:edit
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@my.dokku:edit'
```

So I'm guessing my error is the reverse of what the other user was seeing (so in theory if I was to checkout an earlier commit the error would go away?)

**Comment by dmnobunaga** [comment]
Same error :( Can't build gogs on dokku.

**Comment by dmnobunaga** [comment]
Fixed:
So i revert https://github.com/gogits/gogs/commit/2bec8a4f1ef6b48220a0010c1482e561abd55810 that commit back and everything build and run fine.

**Comment by atsepkov** [comment]
Yep, I ended up reverting to that commit too to get it to work. Encountered an unrelated problem though (nginx 502), which is why I haven't updated this post with my workaround. Ultimately I ended up installing from a docker image (`docker pull gogs/gogs`), which worked for me. I still think there is something in the docker/build.sh script that should be updated to match up with the interface change to solve this bug, but I'm not familiar enough with the config.
